crates/re_log_types/src/component_types/arrow_convert_shims.rs

```diff
@@ -0,0 +1,107 @@
+//! Assorted shims that should make their way back to [arrow2-convert](https://github.com/DataEngineeringLabs/arrow2-convert/)
+
+use std::ops::Index;
+
+use arrow2::{
+    array::{Array, BinaryArray},
+    buffer::Buffer,
+};
+use arrow2_convert::{
+    deserialize::{ArrowArray, ArrowDeserialize},
+    ArrowField, ArrowSerialize,
+};
+
+/// Shim to enable zero-copy arrow deserialization for `Buffer<u8>`
+/// Can be removed when: [arrow2-convert#103](https://github.com/DataEngineeringLabs/arrow2-convert/pull/103) lands
+#[derive(Clone, Debug, PartialEq, ArrowField, ArrowSerialize)]
+#[arrow_field(transparent)]
+pub struct BinaryBuffer(pub Buffer<u8>);
```

Comment:
Isn't `ByteBuffer` a more fititng name name? 😬

Reply:
Arrow calls it a `BinaryArray` so this is the Buffer that a binary array deserializes into.